src/model_repository_manager.cc

```diff
@@ -839,7 +940,17 @@ ModelRepositoryManager::GetModel(
     const std::string& model_name, const int64_t model_version,
     std::shared_ptr<Model>* model)
 {
-  Status status = model_life_cycle_->GetModel(model_name, model_version, model);
+  ModelIdentifier model_id("", model_name);
+  RETURN_IF_ERROR(find_identifier_fn_(model_name, &model_id));
+  return GetModel(model_id, model_version, model);
```

Comment:
Why is the behavior different for here versus the other GetModel below this? That one resets the model if there is an error, this one does not.

Reply:
I think the former one gets model by `model_name`, which could collide between different model namespaces. In the event of a collision, returning an error seems appropriate. The latter one gets model by `model_id`, which uniquely identifies the model. If the model is not found (or not ready), then the `shared_ptr` holding the `Model` gets reset. This should prevent the caller from interacting with the `Model` object, but I am not sure if this is @GuanLuo's intent for the reset.

Reply:
This function calls the `GetModel()` overload below, so it adds additional global lookup on top of the other overload. The `reset()` below is sanity check that we don't care anything over `model_life_cycle_->GetModel()` in case of error (i.e. accidentally set by lifecycle).